src/modules/getBullsAndCows.js

```diff
@@ -12,8 +14,24 @@
  * @return {object} An object containing the number of bulls and cows.
  * Example: { bulls: 1, cows: 2 }
  */
-function getBullsAndCows(userInput, numberToGuess) {
-  /* Write your code here */
+function getBullsAndCows(userInput, numberToGuess = generateRandomNumber()) {
+  let bulls = 0;
+  let cows = 0;
+  const userInputStr = userInput.toString();
+  const numberToGuessStrArr = numberToGuess.toString().split('');
+
+  for (let i = 0; i < 4; i++) {
+    if (numberToGuessStrArr.includes(userInputStr[i])) {
+      cows++;
+    }
+
+    if (userInputStr[i] === numberToGuessStrArr[i]) {
+      bulls++;
+      cows--;
+    }
+  }
```

Comment:
This could be optimized; currently, its complexity is O(n^2), which is okay in this case because n = 4. However, if we increased the number of digits a guess could have, this function would end up being too inefficient. It is possible to solve this in O(n) with a two-pass solution without too many changes to your implementation (though it would also be possible to achieve a one-pass solution).

As an additional (optional) challenge, could you come up with a more efficient solution?